generators/base/generator-base-private.mjs

```diff
@@ -166,11 +162,8 @@ export default class PrivateBase extends Generator {
    * @param folder
    */
   removeFolder(folder) {
-    folder = this.destinationPath(folder);
-    if (folder && shelljs.test('-d', folder)) {
-      this.log(`Removing the folder - ${folder}`);
-      shelljs.rm('-rf', folder);
-    }
+    // TODO Should not update variable
+    folder = deleteFolder(this, folder);
```

Comment:
I don’t think extracting a simple implementation like this into a non reusable lib is better.
IMO makes much more complicated.

Reply:
At least it highlights all the bad practices that were used ^^ (note that assignment is mandatory to get it working here :-S).

The goal behind this is to extract  the logic outside the generator to be able to rewrite the API in a much simpler way: there is a lot of duplicated and reusable code hidden within these 'simple implementations'

Reply:
I agree to extract logic outside the generator, but in this case removeFolder IMO is a generator basic operation.
It's basically the same that yeoman-generator `removeDestination` provides with the difference that this one removes from the physical disk, while the yeoman implementation removes from mem-fs and will be committed latter.

Reply:
This serie of PR will never end and the result won't be that much readable if the guideline is not a binary one (logic == `logic` folder, generator file == public API for blueprint).